src/main/ruby/core/string.rb

```diff
@@ -179,7 +180,8 @@ def rpartition(pattern)
       end
 
       # Nothing worked out, this is the default.
-      ['', '', self]
+      empty = ''.encode(encoding)
```

Comment:
Would `String.new(encoding: encoding)` work here?

Reply:
Most likely. I keep forgetting that new form exists. I'll give it a shot.

Reply:
You would give up the ability to no-op encode on a String literal. But I really don't know that it's worth much of anything here.